pkg/proxy/iptables/proxier.go

```diff
@@ -918,17 +894,21 @@ func (proxier *Proxier) syncProxyRules(reason syncReason) {
 	}
 
 	// Figure out the new services we need to activate.
+	proxier.changeMu.Lock()
 	serviceSyncRequired, hcServices, staleServices := updateServiceMap(
 		proxier.serviceMap, &proxier.serviceChanges)
+	proxier.changeMu.Unlock()
```

Comment:
I think it is safer to lock both updateServiceMap and updateEndpointsMap. I cannot think of any corner case on top of my head. But I feel like syncProxyRule should strictly act on a specific state in sequence. So do not process changes for endpoints after serviceMap is settled.